File: app/src/test/java/com/mercandalli/tracker/main_thread/MainThreadPostImplTest.java
package com.mercandalli.tracker.main_thread;

import android.os.Handler;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

/**
 * Test class for the {@link MainThreadPostImpl}
 */
public class MainThreadPostImplTest {

    /**
     * To test.
     */
    private MainThreadPostImpl mainThreadPostImpl;

    @Mock
    private Handler mainThreadHandler;

    @Before
    public void setup() {
        MockitoAnnotations.initMocks(this);
        mainThreadPostImpl = new MainThreadPostImpl(
                Thread.currentThread(),
                mainThreadHandler);
    }

    @Test
    public void isOnMainThread() {
        Assert.assertTrue(mainThreadPostImpl.isOnMainThread());
    }
}

File: app/src/test/java/com/mercandalli/tracker/main_thread/MainThreadPostImplTest.kt
package com.mercandalli.tracker.main_thread

import android.os.Handler

import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.MockitoAnnotations

/**
 * Test class for the [MainThreadPostImpl]
 */
class MainThreadPostImplTest {

    /**
     * To test.
     */
    private var mainThreadPostImpl: MainThreadPostImpl? = null

    @Mock
    private val mainThreadHandler: Handler? = null

    @Before
    fun setup() {
        MockitoAnnotations.initMocks(this)
        mainThreadPostImpl = MainThreadPostImpl(
                Thread.currentThread(),
                mainThreadHandler!!)
    }

    @Test
    fun isOnMainThread() {
        Assert.assertTrue(mainThreadPostImpl!!.isOnMainThread)
    }
}
